cpp/src/arrow/type_traits.h

```diff
@@ -991,4 +991,20 @@ static inline bool is_nested(Type::type type_id) {
   return false;
 }
 
+static inline int offset_bit_width(Type::type type_id) {
+  switch (type_id) {
+    case Type::STRING:
+    case Type::BINARY:
+    case Type::LIST:
+      return 32;
+    case Type::LARGE_STRING:
+    case Type::LARGE_BINARY:
+    case Type::LARGE_LIST:
+      return 64;
```

Comment:
May want to add `MAP` and `DENSE_UNION` here?

Reply:
ah, good catch